### ISSUE TITLE
Scons build system removes duplicate flags at link time

### ISSUE THREAD

**Issue by ryandesign** [thezbyg/gpick#160]
Using the scons build system, building the latest gpack code in the repo fails if I try to build universal (i.e. for multiple architectures) on macOS. This is accomplished by including e.g. `-arch x86_64 -arch i386` in `CFLAGS`, `CXXFLAGS`, and `LDFLAGS`.

Most of the build succeeds, but it fails linking with:

```
clang: error: no such file or directory: 'x86_64'
```

This appears to be because the build system has removed what it thinks are duplicate flags for the link command. The command that got invoked was:

```
/usr/bin/clang++ -o build/source/gpick -L/opt/local/lib -lexpat x86_64 -arch i386 -Wl,-framework -Wl,CoreFoundation ....
```

Note that the first `-arch` from my specified flags `-arch x86_64 -arch i386` is absent.

**Comment by thezbyg** [comment]
Should be fixed by a40cce8a24f51157d9633fc2c2d436d58cf95c47.
Please test and report back.

**Comment by ryandesign** [comment]
Thanks! It's fixed.
